tasks/kernel_matrix_testing/README.md

```diff
@@ -1,425 +1,564 @@
+
+-   Created
+-   Configured
+-   Launched
+-   Paused
+-   Resumed
+-   Destroyed
+
+### VMsets File
+
+Configuring a stack involves generating a configuration file, referred to as the `vmsets` file. This JSON file specifies the sets of VMs to launch. A "set" of VMs is a collection of VMs sharing characteristics like architecture or purpose (e.g., "all x86_64 distribution VMs on a remote machine").
+
+This file is managed by the `kmt.gen-config` command. You typically don't need to edit it manually.
 
-## Tasks
+A sample VMSet file can be found [here](https://github.com/DataDog/test-infra-definitions/blob/f85e7eb2f003b6f9693c851549fbb7f3969b8ade/scenarios/aws/microVMs/sample-vm-config.json).
 
-### Initializing environment
+### Local vs. Remote VMs
 
-If you only want to initialize the directory structure, you may do a light setup without any parameters:
+KMT can manage two types of VMs:
+
+-   **Local VMs**: These run directly on your machine using `libvirt` and `qemu`. They will have the same architecture as your host machine.
+-   **Remote VMs**: These are EC2 instances launched in AWS. KMT can manage both x86_64 and arm64 remote VMs. Launching remote VMs requires configuring an SSH key.
+
+## Command Reference
+
+This section provides a detailed reference for all `kmt` tasks.
+
+### `kmt.init`
+
+Initializes the KMT environment. It downloads VM images and tools, and installs system dependencies. This command only needs to be run once.
 
 ```bash
-dda inv -e kmt.init
+# Initialize with specific VM images
+dda inv -e kmt.init --images=ubuntu_22.04,debian_10
+
+# Initialize and download all available VM images for your architecture
+dda inv -e kmt.init --all-images
+
+# If you only intend to manage remote VMs
+dda inv -e kmt.init --remote-setup-only
 ```
 
-### Updating resources
+The `--images` parameter is required unless `--remote-setup-only` or `--all-images` is specified, or you explicitly confirm the interactive prompt. Use `kmt.ls` to see available images.
+
+This command will also guide you through the [default SSH key configuration](#kmtconfig-ssh-key).
+
+### `kmt.update-resources`
 
-In order to update the resources for launching VMs run:
+Updates the resources for launching VMs, such as VM images and tools.
 
 ```bash
 dda inv -e kmt.update-resources
 ```
 
-Updating will first destroy all running stacks, and then use checksums to decide which packages need to be updated from S3.
-If there is an error during update, original packages are restored from backup.
+This command will first destroy all running stacks, then use checksums to decide which packages need to be updated from S3. Original packages are restored from a backup if an error occurs.
 
-### Creating a stack
+### `kmt.ls`
 
-A stack can be created as follows:
+Lists available VM images.
+
+```bash
+dda inv -e kmt.ls
+```
+
+Arguments:
+
+-   `--distro`: Only list distribution images (default).
+-   `--custom`: Only list custom kernel images.
+
+### `kmt.create-stack`
```

Comment:
Maybe we can remove this task, and remove it in subsequent PRs. This is almost never used by anyone.

Reply:
Let me keep the documentation for now. I'll open a PR to remove it both from README and the code. I'll post the PR on the internal channel asking if anyone has concerns about this command being removed before merging it.